### PR TITLE
Move conservation check to src

### PR DESCRIPTION
**Pull request by Sbozzolo**
So that it can be used without the driver

### CODE REVIEW

File: src/solver/solve.jl [hunk 1]
```diff
@@ -196,3 +196,69 @@ function precompile_callbacks(integrator)
...
+    else
+        energy_surface_change = -p.net_energy_flux_sfc[][]
+    end
+    energy_radiation_input = -p.net_energy_flux_toa[][]
+
+    energy_net =
+        abs(
+            energy_atmos_change + energy_surface_change -
+            energy_radiation_input,
+        ) / energy_total
```
Comment (szy21) [inline src/solver/solve.jl:242]:
```suggestion
    energy_conservation =
        abs(
            energy_atmos_change + energy_surface_change -
            energy_radiation_input,
        ) / energy_total
```
(I think there is something wrong with the conservation test before? We are dividing it by energy_total here, but then divide it by energy_total again for energy_conservation.)

File: src/solver/solve.jl [hunk 2]
```diff
@@ -196,3 +196,69 @@ function precompile_callbacks(integrator)
...
+        water_atmos_change = sum(sol.u[end].c.ρq_tot) - sum(sol.u[1].c.ρq_tot)
+        water_surface_change = horizontal_integral_at_boundary(
+            sol.u[end].sfc.water .- sol.u[1].sfc.water,
+        )
+
+        water_conservation =
+            abs(water_atmos_change + water_surface_change) / water_total
+    end
+
+    energy_conservation = energy_net / energy_total
```
Comment (szy21) [inline src/solver/solve.jl:261]:
```suggestion
```

File: examples/hybrid/driver.jl [hunk 3]
```diff
@@ -116,55 +116,23 @@ end
 
 # Conservation checks
 if config.parsed_args["check_conservation"]
-    @info "Checking conservation"
     FT = Spaces.undertype(axes(sol.u[end].c.ρ))
+    @info "Checking conservation"
+    (; energy_conservation, mass_conservation, water_conservation) =
+        CA.check_conservation(sol)
+
+    @info "    Net energy change / total energy: $energy_conservation"
```
Comment (szy21) [inline examples/hybrid/driver.jl:124]:
```suggestion
    @info "    Net energy change / total energy: $energy_conservation"
    @info "    Net mass change / total mass: $mass_conservation"
```

**Review by szy21 — APPROVED** [review]
Thanks!

File: src/solver/solve.jl (hunk 1 above)
Comment (Sbozzolo) [reply src/solver/solve.jl]: Yes, I think you are right

**Pull request merged**